shipping_addresses_service.go

```diff
@@ -0,0 +1,82 @@
+package recurly
+
+import (
+	"encoding/xml"
+	"fmt"
+)
+
+var _ ShippingAddressesService = &shippingAddressesImpl{}
+
+// shippingAddressessImpl handles communication with the shipping address
+// related methods of the recurly API.
+type shippingAddressesImpl struct {
+	client *Client
+}
+
+// ListAccount returns a list of all shipping addresses associated with an account.
+func (s *shippingAddressesImpl) ListAccount(accountCode string, params Params) (*Response, []ShippingAddress, error) {
+	action := fmt.Sprintf("accounts/%s/shipping_addresses", accountCode)
+	req, err := s.client.newRequest("GET", action, params, nil)
+	if err != nil {
+		return nil, nil, err
+	}
+
+	var v struct {
+		XMLName           xml.Name          `xml:"shipping_addresses"`
+		ShippingAddresses []ShippingAddress `xml:"shipping_address"`
+	}
+	resp, err := s.client.do(req, &v)
+	return resp, v.ShippingAddresses, err
+}
+
+// Create creates a new shipping address.
+func (s *shippingAddressesImpl) Create(accountCode string, shippingAddress ShippingAddress) (*Response, *ShippingAddress, error) {
+	action := fmt.Sprintf("accounts/%s/shipping_addresses", accountCode)
+	req, err := s.client.newRequest("POST", action, nil, shippingAddress)
+	if err != nil {
+		return nil, nil, err
+	}
+	var sa ShippingAddress
+	resp, err := s.client.do(req, sa)
```

Comment:
`&sa` -- this will currently return a pointer to the zero value of the `ShippingAddress` struct